android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java

```diff
@@ -634,6 +634,10 @@ private String makeKeyboardPath(String packageID, String keyboardID, String keyb
   }
 
   private void sendKMWError(int lineNumber, String sourceId, String message) {
+    if (this.packageID == null || this.keyboardID == null || this.keyboardName == null || this.keyboardVersion == null) {
```

Comment:
Wouldn't it be better to only add the additional data if the field is not null (lines 659-662), rather than dropping the whole event?